db/migrate/20150116095004_patch_corrupt_attachments.rb

```diff
@@ -0,0 +1,48 @@
+##
+# Goes through all attachments looking for ones whose 'file' column, which is the new column
+# used by the carrierwave-based attachments, is not set.
+#
+# For every one of those attachments the migration then sets the 'file' column to
+# whatever the value of the legacy column 'filename' is. If that one is empty too
+# it falls back to the 'disk_filename' column. This one was not meant to be displayed
+# to users but it's better than nothing, especially when trying to identify corrupt attachments.
+#
+# If *that* column is empty too, the attachment is broken beyond repair and will be dropped.
+#
+# Note: Just because the 'file' column is restored doesn't mean the actual file exists.
+#       Rather the 'file' column being empty means precisely that the file is missing.
+#       By still writing the filename into the file column the attachment can at least
+#       be displayed, if not downloaded.
+#
+# Important: The migration is irreversible.
+class PatchCorruptAttachments < ActiveRecord::Migration
+  def up
+    Attachment.all.each do |attachment|
+      patch_attachment attachment
+    end
+  end
+
+  def down
+    puts "Can't revert this migration as it would mean breaking valid attachments."
+    puts "Even if we wanted to do that we can't know which ones were broken before"
+    puts "to only break those again."
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.